pkg/dedicatedadmin/dedicatedadmin_test.go

```diff
@@ -0,0 +1,30 @@
+package dedicatedadmin
+
+import (
+	"testing"
+)
+
+// TestBlacklist exercises the comma-separating in IsBlackListedNamespace
+func TestBlacklist(t *testing.T) {
+	var tests = []struct {
+		configmapstring string // what we might expect from the ConfigMap
+		challenge       string // what would the test namespace be against the regexp?
+		valid           bool   // would we expect it to be a real regexp?
+	}{
+		{"^kube-.*,^openshift-.*,^logging$,^default$,^openshift$", "openshift-test", true},
```

Comment:
Rather we pull the actual config that is in the repo as the basis for a set of tests.  Can pull from `manifests/50-dedicated-admin-operator-config.ConfigMap.yaml` though the name of the file could change.  Maybe symlink to it (assumes we put tests in a dedicated place, not along with the operator src).

Tests I would like to see:

- namespace is exactly "default"
- namespace starts with "default" but is not equal to "default"
- namespace has "default" in the middle or end
- namespace is exactly "openshift"
- namespace starts with "openshift"
- namespace has "openshift" in the middle or end
- namespace is exactly "kube"
- namespace starts with "kube"
- namespace has "kube" in the middle or end
- namespace does not contain strings "kube", "openshift", or "default"

Additional tests could be added with other regex in case we have changes in cluster to the config map.

Reply:
+1 on testing multiple scenarios, I'd suggest using a table-driven approach with the difference namespaces and the expected outcome (blacklisted: true/false). A good example is here: https://github.com/golang/go/wiki/TableDrivenTests

Reply:
@jewzaam The content of the live ConfigMap doesn't matter. Consider the expressions listed in the table to be fixture data that doesn't necessarily need to reflect the real world. I don't want to test `regexp.MatchString` since that doesn't seem a good use of time (we can trust native Go libraries to be well-tested).

@rogbas This is using a table. Maybe you can elaborate what you mean? Related to this, what should the behaviour be if the ConfigMap has `[kube.*` in it? That isn't a valid regular expression (the `[` is mismatched), and will not match `kube-system`. Should the operator fail to start? Something else?

Reply:
Disagree on the contents of the config map.  Maybe it's a separate test to validate the configuration is as expected.  Original configuration for blacklist was incorrect regex that would have matched namespaces like "nmalik-openshift-stuff".  Since the configuration is part of the repository and is deployed from the repository we need to test it.

Reply:
@jewzaam I don't follow what you mean. Can you write a test to illustrate what you mean and propose it as a pull request to my branch (https://github.com/lisa/dedicated-admin-operator/tree/SREP-1037-add-tests ?)

Reply:
https://github.com/lisa/dedicated-admin-operator/pull/1